components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/bindings/handlers/TokenBindingExpiryEventHandler.java

```diff
@@ -0,0 +1,215 @@
+import org.wso2.carbon.identity.application.authentication.framework.context.AuthenticationContext;
+import org.wso2.carbon.identity.application.authentication.framework.model.AuthenticatedUser;
+import org.wso2.carbon.identity.application.authentication.framework.util.FrameworkConstants;
+import org.wso2.carbon.identity.event.IdentityEventConstants;
+import org.wso2.carbon.identity.event.IdentityEventException;
+import org.wso2.carbon.identity.event.event.Event;
+import org.wso2.carbon.identity.event.handler.AbstractEventHandler;
+import org.wso2.carbon.identity.oauth.OAuthUtil;
+import org.wso2.carbon.identity.oauth.common.exception.InvalidOAuthClientException;
+import org.wso2.carbon.identity.oauth2.IdentityOAuth2Exception;
+import org.wso2.carbon.identity.oauth2.dao.OAuthTokenPersistenceFactory;
+import org.wso2.carbon.identity.oauth2.internal.OAuth2ServiceComponentHolder;
+import org.wso2.carbon.identity.oauth2.model.AccessTokenDO;
+import org.wso2.carbon.identity.oauth2.token.bindings.TokenBinder;
+import org.wso2.carbon.identity.oauth2.util.OAuth2Util;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+
+import static org.wso2.carbon.identity.application.authentication.framework.util.FrameworkConstants.COMMONAUTH_COOKIE;
+import static org.wso2.carbon.identity.application.authentication.framework.util.FrameworkConstants.RequestParams.TYPE;
+
+/**
+ * Event handler for token revocation during access token binding expiration.
+ */
+public class TokenBindingExpiryEventHandler extends AbstractEventHandler {
+
+    private static final Log log = LogFactory.getLog(TokenBindingExpiryEventHandler.class);
+    private static final String SSO_SESSION_BINDING = "sso-session";
+
+    @Override
+    public void handleEvent(Event event) throws IdentityEventException {
+
+        if (log.isDebugEnabled()) {
+            log.debug(event.getEventName() + " event received to TokenBindingExpiryEventHandler.");
+        }
+
+        if (StringUtils.equals(event.getEventName(), IdentityEventConstants.EventName.SESSION_TERMINATE.name())) {
+
+            HttpServletRequest request = getHttpRequestFromEvent(event);
+            Map<String, Object> eventProperties = event.getEventProperties();
+            AuthenticationContext context = (AuthenticationContext) eventProperties.get(IdentityEventConstants
+                    .EventProperty.CONTEXT);
+            try {
+                if (StringUtils.equals(request.getParameter(TYPE), FrameworkConstants.RequestType.CLAIM_TYPE_OIDC)) {
+
+                    String consumerKey = context.getRelyingParty();
+                    String bindingType = OAuth2Util.getAppInformationByClientId(consumerKey).getTokenBindingType();
+
+                    revokeTokensForBindingType(request, context.getLastAuthenticatedUser(), consumerKey, bindingType);
```

Comment:
if there are no binding types we can skip this

Reply:
added a null check and called this method